src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -1,192 +1,139 @@
-
-    private FourSparkMaxWithSettings speedModeSparks;
-    private FourSparkMaxWithSettings positiionModeSparks;
+    private CANEncoder frontLeftEncoder;
+    private CANEncoder frontRightEncoder;
+    private CANEncoder rearLeftEncoder;
+    private CANEncoder rearRightEncoder;
 
-    private SparkPositionControllerGroup posController;
-
-    private EncoderPoseGenerator poseGen;
-
-    @Override
-    public void initialize() {
-        m_frontLeft = new CANSparkMax(RobotMap.CAN.FRONT_LEFT_MOTOR, MotorType.kBrushless);
-        m_rearLeft = new CANSparkMax(RobotMap.CAN.REAR_LEFT_MOTOR, MotorType.kBrushless);
-        m_left = new SpeedControllerGroup(m_frontLeft, m_rearLeft);
+    private SparkPositionControllerGroup positionController;
     
-        m_frontRight = new CANSparkMax(RobotMap.CAN.FRONT_RIGHT_MOTOR, MotorType.kBrushless);
-        m_rearRight = new CANSparkMax(RobotMap.CAN.REAR_RIGHT_MOTOR, MotorType.kBrushless);
-        m_right = new SpeedControllerGroup(m_frontRight, m_rearRight);
-        
-
-        m_robotDrive = new DifferentialDrive(m_left, m_right);
-
-        SparkMaxSettings frontLeftSpeedSettings = SparkMaxSettingsBuilder.defaults()
-                                                    .withCurrentLimits(35)
-                                                    .coastInNeutral()
-                                                    .withDirections(false, false)
-                                                    .limitMotorOutputs(1.0, -1.0)
-                                                    .noMotorStartupRamping()
-                                                    .useSpeedControl()
-                                                    .build();
-                                                    
-                                                
-        SparkMaxSettings frontRightSpeedSettings = SparkMaxSettingsBuilder.defaults()
-                                                    .withCurrentLimits(35)
-                                                    .coastInNeutral()
-                                                    .withDirections(false, false)
-                                                    .limitMotorOutputs(1.0, -1.0)
-                                                    .noMotorStartupRamping()
-                                                    .useSpeedControl()
-                                                    .build();
+    private SpeedControllerGroup leftSpeedController;
+    private SpeedControllerGroup rightSpeedController;    
+    private DifferentialDrive robotDrive;    
 
-
-
-        SparkMaxSettings rearLeftSpeedSettings = SparkMaxSettingsBuilder.defaults()
+    private SparkMaxSettings speedSettings = SparkMaxSettingsBuilder.defaults()
                                                     .withCurrentLimits(35)
                                                     .coastInNeutral()
                                                     .withDirections(false, false)
```

Comment:
i think we have learned that you can't reverse the encoders for sparkmaxes. therefore, it shouldnt be possible to try.